CA_DataUploaderLib/BaseSensorBox.cs

```diff
@@ -487,21 +487,25 @@ private async Task ReadSensors(MCUBoard board, string boxName, SensorSample.Inpu
                         ProcessLine(numbers, board, targetSamples);
                         lastValidReadTime = _cmd.Time.GetTimestamp();
                         _boardsState.SetState(boxName, ConnectionState.ReceivingValues);
+
+                        if (status != _lastStatus && (status & 0x01000000) != 0) // Flash ongoing?

```

Comment:
Do you need the comparison `!=0` here? I think it would evaluate to 1 if it is set. Same for the last comparison in line 493. 

Otherwise it looks good :=)

Reply:
it's either that or casting/converting it to a bool - I like this better :)

Reply:
Wouldn't an int evaluating to anything but 0 pass as true? That's how it is in C, but maybe it is different for you. Anyway also not something I'm really concerned with just a suggestion. :smile:

Reply:
different in C# :)
![image](https://github.com/user-attachments/assets/96a0f53d-0126-4f60-8eda-8f265e6253d9)